language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs

```diff
@@ -193,55 +174,65 @@ tests damlTypes yarn damlc daml2ts davl = testGroup "daml2ts tests"
           , davl </> "davl-v5.dar"
           , davl </> "davl-upgrade-v4-v5.dar" ] ++
           ["-o", daml2tsDir]
-        assertFileExists (here </> "package.json")
-        -- Overwrite the 'package.json' that daml2ts generated because
-        -- we need to adjust module resolution for @daml/types and
-        -- @daml/ledger.
-        setupWorkspace
-        -- Call daml2ts again which will this time update 'package.json'.
-        callProcessSilent daml2ts $
-          [ davl </> "davl-v4.dar"
-          , davl </> "davl-v5.dar"
-          , davl </> "davl-upgrade-v4-v5.dar" ] ++
-          ["-o", daml2tsDir] -- There's no need to pass '-p
-                             -- here/package.json' but we could and it would mean the same.
         assertFileExists (daml2tsDir </> "davl-0.0.4" </> "src" </> "DAVL.ts")
-        assertFileExists (daml2tsDir </> "davl-0.0.5" </> "src" </> "DAVL.ts")
-        assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "src" </> "Upgrade.ts")
-        step "yarn install..."
-        yarnProject ["install"]
-        step "yarn workspaces run build..."
-        yarnProject ["workspaces", "run", "build"]
         assertFileExists (daml2tsDir </> "davl-0.0.4" </> "lib" </> "DAVL.js")
+        assertFileExists (daml2tsDir </> "davl-0.0.4" </> "lib" </> "DAVL.d.ts")
+        assertFileExists (daml2tsDir </> "davl-0.0.5" </> "src" </> "DAVL.ts")
         assertFileExists (daml2tsDir </> "davl-0.0.5" </> "lib" </> "DAVL.js")
+        assertFileExists (daml2tsDir </> "davl-0.0.5" </> "lib" </> "DAVL.d.ts")
+        assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "src" </> "Upgrade.ts")
         assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "lib" </> "Upgrade.js")
-        step "yarn workspaces run lint..."
-        yarnProject ["workspaces", "run", "lint"]
-     ]
+        assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "lib" </> "Upgrade.d.ts")
+      step "eslint..."
+      withCurrentDirectory daml2tsDir $ do
+        pkgs <- (\\ ["package.json", "node_modules"]) <$> listDirectory daml2tsDir
+        BSL.writeFile "package.json" $ encode (
+          object
+            [ "private" .= True
+            , "devDependencies" .= HMS.fromList
+              ([ ("eslint", "^6.7.2")
+               , ("@typescript-eslint/eslint-plugin", "2.11.0")
+               , ("@typescript-eslint/parser", "2.11.0")
+               ] :: [(T.Text, T.Text)]
+              )
+            , "dependencies" .= HMS.fromList
+              ([ ("@daml/types", "file:../daml-types")
+               , ("@mojotech/json-type-validation", "^3.1.0")
+               ] :: [(T.Text, T.Text)])
+            , "workspaces" .= pkgs
+            , "name" .= ("daml2ts" :: T.Text)
+            , "version" .= ("0.0.0" :: T.Text)
+            ])
+        callProcessSilent yarn ["install", "--pure-lockfile"]
+        callProcessSilent yarn ["workspaces", "run", "lint"]
+  ]
```

Comment:
I have a suspicion that this only works because we write too much in the `package.json` file for every package. I'm fine with keeping it in there for now and remove it in a followup PR.

Reply:
I hope your suspicions are unfounded.
```haskell

    packageJson :: Value
    packageJson = object
      [ "name" .= ("daml2ts" :: T.Text)
      , "version" .= version
      , "dependencies" .= pkgDependencies config
      , "devDependencies" .= pkgDevDependencies config
      ]
```
where,
```haskell
configConsts :: T.Text -> ConfigConsts
configConsts sdkVersion = ConfigConsts
  { pkgDependencies = HMS.fromList
      [ (NpmPackageName "@mojotech/json-type-validation", NpmPackageVersion "^3.1.0")
      , (NpmPackageName "@daml/types", NpmPackageVersion sdkVersion)
      ]
  , pkgDevDependencies = HMS.fromList
      [ (NpmPackageName "typescript", NpmPackageVersion "~3.7.3")
      ]
  , pkgScripts = HMS.fromList
      [ (ScriptName "build", Script "tsc")
      , (ScriptName "lint", Script "eslint --ext .ts --max-warnings 0 src/")
      ]
  }
```
Let me know!

Reply:
I do observe that the `pkgDependencies` are not required for the lint test. Removing those in this PR.

Reply:
Oh wait! I think you mean we do something similar in the DAML assistant too. Yes, we should definitely factor this out then. I'll put it on the stack with the 2 notes below.

Reply:
Does that mean we can delete `pkgScripts`? Let's do that then! 🔥

Reply:
Done.